src/eos-autoupdater.c

```diff
@@ -599,10 +601,29 @@ is_time_to_update (guint update_interval_days,
      * computer’s run eos-autoupdater. In order to avoid a thundering herd of
      * computers requesting updates when a lab is first turned on, create a
      * stamp file with a random delay applied, and check again for updates
-     * later. */
-    g_debug ("Not time to update, due to stamp file not being present.");
-    update_stamp_file (update_interval_days, randomized_delay_days);
-    is_time_to_update = FALSE;
+     * later. To do this, we fake the date of the most recent successful update
+     * to be @update_interval_days in the past, so only the
+     * @randomized_delay_days is taken into account for triggering the next
+     * update. */
+    if (randomized_delay_days > 0)
+      {
+        guint64 last_successful_update_secs;
+
+        g_debug ("Not time to update, due to stamp file not being present, but %s is set to %u days.",
+                 RANDOMIZED_DELAY_KEY, randomized_delay_days);
+        last_successful_update_secs = g_get_real_time () / G_USEC_PER_SEC;
+        if (last_successful_update_secs >= update_interval_days * SEC_PER_DAY)
+          last_successful_update_secs -= update_interval_days * SEC_PER_DAY;
```

Comment:
I'm not sure I completely understand the issue here; since we're setting the mtime of the stamp file `IntervalDays` in the past, if the interval was originally one week and then the user changed it to two weeks, would we not start an update one week after the change? Would you expect that the update happens two weeks after the change at that point instead?